frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostStatus.tsx

```diff
@@ -64,4 +70,8 @@ const BareMetalHostStatus: React.FC<StatusProps> = ({ status, title, description
   }
 };
 
+type BareMetalHostStatusProps = StatusProps & {
```

Comment:
I think we should keep the StatusProps in tact and specify it further here. The host should be optional as it is not necessary in all states

Reply:
OK, but I removed the `[key: string]: any` from `StatusProps` which basically removes any type-checking :) and added optional `nodeMaintenance` to both BMN and BMH statuses

Reply:
hmm, the goal of that was that status object has status, title, description and any amount of additional data which can be used by the status components so it does not have to be passed there explicitly and it is already included in the status object. With this in place, you can just do <BareMetalHostStatus {...status} /> without explicitly passing host and maintenance objects because they are already included in status object.